src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -2435,7 +2435,8 @@ else if (!useTnir) {
 
         }
 
-        assert result.equals(Result.SUCCESS) == true;
+        boolean equalSuccess = result.equals(Result.SUCCESS);
```

Comment:
SonarQube might happy but I don't see any usage of `equalSuccess` other than assert. 

I guess following code will be more useful.

```
assert (result.equals(Result.SUCCESS))
```
**Think about this in different way:** 
What will happen if result is false **if assertions are not enabled**. 

Are we handle this in somewhere (in our exception flow)? Should we want to handle it here?